Formula/xidel.rb

```diff
@@ -0,0 +1,77 @@
+class Xidel < Formula
+  desc "XPath/XQuery 3.0, JSONiq interpreter to extract data from HTML/XML/JSON"
+  homepage "https://www.videlibri.de/xidel.html"
+  license "GPL-3.0-or-later"
+
+  class TargetedResource < Resource
+    def initialize(*rest, **kw_rest)
+      @target = nil
+      super(*rest, **kw_rest)
+    end
+
+    def target(val = nil)
+      return @target if val.nil?
+
+      @target = val
+    end
+  end
+
+  stable do
+    url "https://github.com/benibela/xidel/releases/download/Xidel_0.9.8/xidel-0.9.8.src.tar.gz"
+    sha256 "72b5b1a2fc44a0a61831e268c45bc6a6c28e3533b5445151bfbdeaf1562af39c"
+  end
+
+  livecheck do
+    url :stable
+    strategy :github_latest
+    regex(%r{href=.*?/tag/Xidel_(\d+(?:\.\d+)+)["' >]}i)
+  end
+
+  head do
+    url "https://github.com/benibela/xidel.git", branch: "master"
+
+    resource "flre", TargetedResource do
+      url "https://github.com/benibela/flre.git", branch: "master"
+      target "import/#{name}"
+    end
+    resource "internettools", TargetedResource do
+      url "https://github.com/benibela/internettools.git", branch: "master"
+      target "import/#{name}"
+    end
+    resource "pasdblstrutils", TargetedResource do
+      url "https://github.com/BeRo1985/pasdblstrutils.git", branch: "master"
+      target "import/#{name}"
+    end
+    resource "rcmdline", TargetedResource do
+      url "https://github.com/benibela/rcmdline.git", branch: "master"
+      target "import/#{name}"
+    end
+    resource "synapse", TargetedResource do
+      url "http://svn.code.sf.net/p/synalist/code/synapse/40"
+      target "import/#{name}"
+    end
+  end
+
+  depends_on "fpc" => :build
+  depends_on "openssl@1.1"
+
+  def install
+    resources.select { |r| r.is_a?(TargetedResource) }.each do |r|
+      mkdir_p (buildpath/r.target).dirname
+      r.stage buildpath/r.target
+    end
+
+    cd File.directory?("programs/internet/xidel") ? "programs/internet/xidel" : "." do
+      inreplace "build.sh" do |s|
+        s.gsub!(/\$fpc /, "\$fpc -k-rpath -k#{sh_quote Formula["openssl@1.1"].opt_lib} ")
```

Comment:
Are you sure escaping the second `$` is necessary? Why not just wrap `Formula["openssl@1.1"].opt_lib` in `'` to prevent shell expansion/word splitting?

Reply:
There's no need to escape if you wrap it in quotes.

Reply:
Single quotes are legal characters in pathnames.

Reply:
They are, but no one puts single quotes in pathnames because it would be ridiculous to work with.

Reply:
Not personally, but that's not an excuse to be fragile. Homebrew is written in Ruby. Calling `sh_quote` is very cheap, and it will always be accurate. Why would using fragile quotes in the string literal be a valuable enough optimization to justify sacrificing correctness?